src/Surfnet/StepupSelfService/SamlStepupProviderBundle/Provider/MetadataFactoryCollection.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types = 1);
+
+/**
+ * Copyright 2013 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupSelfService\SamlStepupProviderBundle\Provider;
+
+use Surfnet\SamlBundle\Metadata\MetadataFactory;
+use Surfnet\StepupSelfService\SamlStepupProviderBundle\Exception\MetadataFactoryNotFoundException;
+
+class MetadataFactoryCollection
+{
+    /**
+     * @var MetadataFactory[]
```

Comment:
Can be more specifically defined like `@var array<string, MetadataFactory>`. That way you also express the key is always of type string.